modules/standard/IO.chpl

```diff
@@ -3754,12 +3802,13 @@ private inline proc _write_one_internal(_channel_internal:qio_channel_ptr_t,
 pragma "no doc"
 proc channel.readIt(ref x) {
   if writing then compilerError("read on write-only channel");
+  const origLocale = this.getIoStartedOnLocale();
   on this.home {
     this.lock();
     var error:syserr;
     error = qio_channel_error(_channel_internal);
     if ! error {
-      error = _read_one_internal(_channel_internal, kind, x);
+      error = _read_one_internal(_channel_internal, kind, x, origLocale);
```

Comment:
The trivial answer is that in that function, `origLocale` is set by `const origLocale = this.getIoStartedOnLocale();`. But the answer you were probably looking for is that _readWriteThisFromLocale is set in _read_one_internal and _write_one_internal overloads that handle non-primitive types in the construction of a new, unlocked channel. Search for _readWriteThisFromLocale to find it.